language/move-core/types/src/gas_schedule.rs

```diff
@@ -0,0 +1,266 @@
+define_gas_unit! {
+    name: AbstractMemorySize,
+    carrier: GasCarrier,
+    doc: "A newtype wrapper that represents the (abstract) memory size that the instruction will take up."
+}
+
+define_gas_unit! {
+    name: GasUnits,
+    carrier: GasCarrier,
+    doc: "A newtype wrapper around the underlying carrier for the gas cost."
+}
+
+define_gas_unit! {
+    name: GasPrice,
+    carrier: GasCarrier,
+    doc: "A newtype wrapper around the gas price for each unit of gas consumed."
+}
+
+/// Zero cost.
+pub const ZERO_GAS_UNITS: GasUnits<GasCarrier> = GasUnits(0);
+
+/// The cost per-byte written to global storage.
+/// TODO: Fill this in with a proper number once it's determined.
+pub const GLOBAL_MEMORY_PER_BYTE_COST: GasUnits<GasCarrier> = GasUnits(8);
+
+/// The cost per-byte written to storage.
+/// TODO: Fill this in with a proper number once it's determined.
+pub const GLOBAL_MEMORY_PER_BYTE_WRITE_COST: GasUnits<GasCarrier> = GasUnits(8);
+
+/// The maximum size representable by AbstractMemorySize
+pub const MAX_ABSTRACT_MEMORY_SIZE: AbstractMemorySize<GasCarrier> =
+    AbstractMemorySize(std::u64::MAX);
+
+/// The units of gas that should be charged per byte for every transaction.
+pub const INTRINSIC_GAS_PER_BYTE: GasUnits<GasCarrier> = GasUnits(8);
+
+/// The minimum gas price that a transaction can be submitted with.
+pub const MIN_PRICE_PER_GAS_UNIT: GasPrice<GasCarrier> = GasPrice(0);
+
+/// The maximum gas unit price that a transaction can be submitted with.
+pub const MAX_PRICE_PER_GAS_UNIT: GasPrice<GasCarrier> = GasPrice(10_000);
+
+/// 1 nanosecond should equal one unit of computational gas. We bound the maximum
+/// computational time of any given transaction at 10 milliseconds. We want this number and
+/// `MAX_PRICE_PER_GAS_UNIT` to always satisfy the inequality that
+///         MAXIMUM_NUMBER_OF_GAS_UNITS * MAX_PRICE_PER_GAS_UNIT < min(u64::MAX, GasUnits<GasCarrier>::MAX)
+pub const MAXIMUM_NUMBER_OF_GAS_UNITS: GasUnits<GasCarrier> = GasUnits(1_000_000);
+
+/// We charge one unit of gas per-byte for the first 600 bytes
+pub const MIN_TRANSACTION_GAS_UNITS: GasUnits<GasCarrier> = GasUnits(600);
+
+/// The word size that we charge by
+pub const WORD_SIZE: AbstractMemorySize<GasCarrier> = AbstractMemorySize(8);
+
+/// The size in words for a non-string or address constant on the stack
+pub const CONST_SIZE: AbstractMemorySize<GasCarrier> = AbstractMemorySize(1);
+
+/// The size in words for a reference on the stack
+pub const REFERENCE_SIZE: AbstractMemorySize<GasCarrier> = AbstractMemorySize(8);
+
+/// The size of a struct in words
+pub const STRUCT_SIZE: AbstractMemorySize<GasCarrier> = AbstractMemorySize(2);
+
+/// For V1 all accounts will be 32 words
+pub const DEFAULT_ACCOUNT_SIZE: AbstractMemorySize<GasCarrier> = AbstractMemorySize(32);
+
+/// Any transaction over this size will be charged `INTRINSIC_GAS_PER_BYTE` per byte
+pub const LARGE_TRANSACTION_CUTOFF: AbstractMemorySize<GasCarrier> = AbstractMemorySize(600);
+
```

Comment:
Is the plan to eventually move these on-chain as well? What are your thoughts on it?

Reply:
I'll move those to on-chain in a separate PR!